frappe/public/js/frappe/widgets/chart_widget.js

```diff
@@ -343,17 +362,11 @@ frappe.ui.DashboardChart = class DashboardChart {
 			colors = [this.chart_doc.color || "light-blue"];
 		}
 
-		this.chart_container.find('.chart-loading-state').addClass('hide');
-		if (!this.data) {
-			this.chart_container.find('.chart-empty-state').removeClass('hide');
+		if (!this.data || !Object.keys(this.data).length) {
+			const empty = $(`<div class="chart-loading-state text-muted" style="height: 200px;">${__("No Data...")}</div>`);
```

Comment:
This should replace the body content even when the chart is updated by applying filters.